### PR TITLE
3.0.0-Chip

### PR DESCRIPTION
**Pull request by CloudPower97**
# Description

This Pull Request is for #588 

It's now possible to use the JavaScript Plugin.

These are the new options added (as per the [documentation](https://materializecss.com/chips.html#options)):

- data
- placeholder
- secondaryPlaceholder
- autocompleteOptions
- limit
- onChipAdd
- onChipSelect
- onChipDelete

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

I have added a new test using the `new-mocker` helper.

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have not generated a new package version. (the maintainers will handle that)

### CODE REVIEW

File: src/Chip.js [hunk 1]
```diff
@@ -1,23 +1,95 @@
...
-      {children}
-      {close ? <i className="close material-icons">close</i> : null}
-    </div>
-  );
-};
+class Chip extends Component {
+  componentDidMount() {
+    const { options } = this.props;
+
+    if (options && M) {
```
Comment (alextrastero) [inline src/Chip.js]: if `options` is not defined, this component shouldn't initialise?

File: src/Chip.js [hunk 2]
```diff
@@ -1,23 +1,95 @@
...
+      {
+        chip: !options,
+        chips: options
+      },
+      className
+    );
+    let chipContent = (
+      <React.Fragment>
+        {children}
+        {close ? <i className="close material-icons">close</i> : null}{' '}
```
Comment (alextrastero) [inline src/Chip.js]: should we add here `<Icon>` component like in others?

Comment (alextrastero) [reply src/Chip.js]: what's the empty string for?

File: src/Chip.js [hunk 3]
```diff
@@ -1,23 +1,95 @@
...
+    );
+
+    if (options) {
+      chipContent = null;
+    }
+
+    return (
+      <div
+        className={classes}
+        {...this.props}
```
Comment (alextrastero) [inline src/Chip.js]: Don't we get a warning for unwanted props on `div` like this?

File: src/Chip.js (hunk 1 above)
Comment (CloudPower97) [reply src/Chip.js]: Nope, as matter of fact you may want to use a `Chip` but not the __[JavaScript plugin](https://materializecss.com/chips.html#basic)__.
Also, the __CSS__ is different too, `chip` when you do not want to use the plugin, `chips` when you want to.

File: src/Chip.js (hunk 2 above)
Comment (CloudPower97) [reply src/Chip.js]: > what's the empty string for?

Whoops, my bad!

File: src/Chip.js (hunk 1 above)
Comment (alextrastero) [reply src/Chip.js]: Cool, ok

**Comment by CloudPower97** [comment]
@alextrastero ping :smile:

**Comment by alextrastero** [comment]
Check out my latest commits @CloudPower97 :)

**Pull request merged**